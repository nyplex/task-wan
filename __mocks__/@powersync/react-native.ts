module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
