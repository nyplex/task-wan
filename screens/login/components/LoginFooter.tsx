import { useRouter } from "expo-router";
import { useSelector } from "react-redux";
import { selectAuthStatus } from "@/redux/slices/authSlice/authSelectors";
import { Box } from "@/components/gluestack/box";
import Text from "@/components/Text";

const LoginFooter = () => {
  const router = useRouter();
  const isLoading = useSelector(selectAuthStatus);

  return (
    <Box>
      <Text className="text-center mt-8">
        Don't have an account?{" "}
        <Text
          weight="bold"
          onPress={() => router.navigate("/(app)/Register")}
          disabled={isLoading}
          className="underline">
          Sign Up
        </Text>
      </Text>
    </Box>
  );
};

export default LoginFooter;
