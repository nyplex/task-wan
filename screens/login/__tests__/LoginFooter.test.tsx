import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import LoginFooter from "../components/LoginFooter";

jest.mock("expo-router", () => ({ useRouter: () => ({ navigate: jest.fn() }) }));
jest.mock("react-redux", () => ({ useSelector: () => false }));

describe("LoginFooter", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON, getByText } = render(<LoginFooter />);
    expect(getByText("Don't have an account? Sign Up")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders Sign Up as bold and underlined", () => {
    const { getByText } = render(<LoginFooter />);
    const signUp = getByText("Sign Up");
    expect(signUp.props.className).toContain("underline");
    expect(signUp.props.className).toContain("font-bold");
  });

  it("calls router.navigate when Sign Up is pressed", () => {
    const mockNavigate = jest.fn();
    jest.spyOn(require("expo-router"), "useRouter").mockReturnValue({ navigate: mockNavigate });
    const { getByText } = render(<LoginFooter />);
    fireEvent.press(getByText("Sign Up"));
    expect(mockNavigate).toHaveBeenCalledWith("/(app)/Register");
  });

  it("disables Sign Up when loading", () => {
    jest.spyOn(require("react-redux"), "useSelector").mockReturnValue(true);
    const { getByText } = render(<LoginFooter />);
    expect(getByText("Sign Up").props.disabled).toBe(true);
  });
});
