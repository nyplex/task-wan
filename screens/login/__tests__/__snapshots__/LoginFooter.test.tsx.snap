// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFooter renders correctly and matches snapshot 1`] = `
<View
  className="flex-row justify-center items-center gap-1"
>
  <Text
    className="based-styles font-regular text-typography-primary text-[16px] tracking-[0px] text-center mt-8"
    testID="login-footer"
  >
    Don't have an account?
     
  </Text>
  <Text
    className="font-body based-styles font-bold text-typography-primary text-[16px] tracking-[0px] underline mt-8"
    disabled={false}
    onPress={[Function]}
    testID="login-footer-signup"
  >
    Sign Up
  </Text>
</View>
`;
